### PR TITLE
fix!: incorrect CalcCbTxMerkleRootQuorums with rotation

### PR DESCRIPTION
**Pull request by ogabrielides**
MerkleRootQuorum calculation for new blocks was handling incorrectly the block containing commitments which are not returned by `GetMinedAndActiveCommitmentsUntilBlock` yet.

Had to remove static variables (caching) because processing got more complex:

The edge case when only some rotated quorums were formed needs to be handled: only the corresponding formed quorum indexes need to be replaced

### CODE REVIEW

**Review by PastaPastaPasta — DISMISSED** [review]
I guess this is fine? But it really concerns me that we don't have tests.

utACK for squash merge

File: src/evo/cbtx.cpp [hunk 1]
```diff
@@ -215,6 +213,12 @@ bool CalcCbTxMerkleRootQuorums(const CBlock& block, const CBlockIndex* pindexPre
             auto qcHash = ::SerializeHash(qc.commitment);
             const auto& llmq_params = llmq::GetLLMQParams(qc.commitment.llmqType);
             auto& v = qcHashes[llmq_params.type];
+            if (llmq::CLLMQUtils::IsQuorumRotationEnabled(qc.commitment.llmqType, pindexPrev)) {
+                auto& qi = qcIndexedHashes[qc.commitment.llmqType];
+                qi.erase(qc.commitment.quorumIndex);
+                qi.insert(std::make_pair(qc.commitment.quorumIndex, qcHash));
```
Comment (UdjinM6) [inline src/evo/cbtx.cpp:219]:
```suggestion
                qi[qc.commitment.quorumIndex] = qcHash;
```
would be a bit more readable imo

**Review by UdjinM6 — COMMENTED** [review]
LGTM, one suggestion

**Comment by ogabrielides** [comment]
@UdjinM6 @PastaPastaPasta Updated

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/test_framework/messages.py [hunk 2]
```diff
@@ -1144,6 +1149,8 @@ def serialize(self):
         r += self.membersSig
         return r
 
+    def __str__(self):
+        return "quorumIndex:{0}, quorumHash:{1}, nVersion:{2}\n".format(self.llmqType, self.quorumHash, self.nVersion)
```
[the remark this answers is not in the document]
Comment (ogabrielides) [reply test/functional/test_framework/messages.py]: You are right

**Comment by ogabrielides** [comment]
@UdjinM6 Updated again

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by QuantumExplorer — DISMISSED** [review]
Approving at the request of @PastaPastaPasta, looked at it, and with my little knowledge, it does look okay.

**Review by PastaPastaPasta — APPROVED** [review]
re-utACK fixed linter

**Pull request merged**